tests/unit/mathematics/linalg/LinalgRefactor_unittest.cc

```diff
@@ -0,0 +1,52 @@
+#include <shogun/lib/config.h>
+
+#include <shogun/mathematics/linalgrefactor/linalgRefactor.h>
+#include <shogun/lib/SGVector.h>
+#include <algorithm>
+#include <memory>
+#include <gtest/gtest.h>
+
+#include <shogun/mathematics/eigen3.h>
+
+using namespace shogun;
+
+/*
+TEST(LinalgRefactor, split_header_and_cpp)
+{
+    const index_t size = 10;
+    SGVector<int32_t> a(size);
+    for (index_t i = 0; i < size; ++i) a[i] = i;
+    GPU_Vector<int32_t> a_GPU(a);
+}
+*/
+
+TEST(LinalgRefactor, CPU_Vector_convert)
+{
+    const index_t size = 10;
+    SGVector<int32_t> a(size);
+    for (index_t i = 0; i < size; ++i) a[i] = i;
+
+    CPUVector<int32_t> a_CPU(a);
+
+    for (index_t i = 0; i < size; ++i)
+        EXPECT_NEAR(a[i], (a_CPU.CPUptr)[i], 1E-15);
+}
+
+TEST(LinalgRefactor, CPUBackend_dot)
+{
+    const index_t size=10;
+    SGVector<int32_t> a(size), b(size);
+    a.set_const(1);
+    b.set_const(2);
+
+    CPUVector<int32_t> a_CPU(a);
+    CPUVector<int32_t> b_CPU(b);
+
+    CPUBackend eigenBackend;
+    Linalg lr;
```

Comment:
No need to create the Linalg object here now. You have already added the global instance of it.